src/graph/network.cc

```diff
@@ -44,13 +54,26 @@ static void RecvData(network::Receiver* receiver,
 DGL_REGISTER_GLOBAL("network._CAPI_DGLSenderCreate")
 .set_body([] (DGLArgs args, DGLRetValue* rv) {
     network::Sender* sender = new network::SocketSender();
+    CommunicatorHandle chandle = static_cast<CommunicatorHandle>(sender);
     try {
       char* buffer = new char[kMaxBufferSize];
-      sender->SetBuffer(buffer);
+      MemoryPool[chandle] = buffer;
     } catch (const std::bad_alloc&) {
       LOG(FATAL) << "Not enough memory for sender buffer: " << kMaxBufferSize;
     }
-    CommunicatorHandle chandle = static_cast<CommunicatorHandle>(sender);
+    *rv = chandle;
+  });
+
+DGL_REGISTER_GLOBAL("network._CAPI_DGLReceiverCreate")
+.set_body([] (DGLArgs args, DGLRetValue* rv) {
+    network::Receiver* receiver = new network::SocketReceiver();
+    CommunicatorHandle chandle = static_cast<CommunicatorHandle>(receiver);
+    try {
+      char* buffer = new char[kMaxBufferSize];
+      MemoryPool[chandle] = buffer;
+    } catch (const std::bad_alloc&) {
+      LOG(FATAL) << "Not enough memory for receiver buffer: " << kMaxBufferSize;
+    }
```

Comment:
did you just move the location of the code?

Reply:
> did you just move the location of the code?

I change the code: `MemoryPool[chandle] = buffer;`